frontend/src/features/VesselSidebar/Reportings/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { useEffect, useState } from 'react'
+import { FingerprintSpinner } from 'react-epic-spinners'
+import styled from 'styled-components'
+
+import { Archived } from './Archived'
+import { Current } from './Current'
+import { COLORS } from '../../../constants/constants'
+import { vesselsAreEquals } from '../../../domain/entities/vessel/vessel'
+import getVesselReportings from '../../../domain/use_cases/vessel/getVesselReportings'
+import { useMainAppDispatch } from '../../../hooks/useMainAppDispatch'
+import { useMainAppSelector } from '../../../hooks/useMainAppSelector'
+import { usePrevious } from '../../../hooks/usePrevious'
+
+const ReportingTab = {
+  CURRENT_REPORTING: 'CURRENT_REPORTING',
+  REPORTING_HISTORY: 'REPORTING_HISTORY'
+}
+
+export function Reportings() {
+  const dispatch = useMainAppDispatch()
+  const { selectedVesselIdentity } = useMainAppSelector(state => state.vessel)
+
+  const { archivedReportingsFromDate, currentAndArchivedReportingsOfSelectedVessel, isLoadingReporting } =
+    useMainAppSelector(state => state.reporting)
+
+  const [reportingTab, setReportingTab] = useState(ReportingTab.CURRENT_REPORTING)
+  const previousSelectedVesselIdentity = usePrevious(selectedVesselIdentity)
+
+  useEffect(() => {
+    if (!archivedReportingsFromDate) {
+      return
+    }
+
+    dispatch(getVesselReportings())
+    if (!vesselsAreEquals(previousSelectedVesselIdentity, selectedVesselIdentity)) {
```

Comment:
That's better to use a `useMemo()` to have only the the data fetching inside this `useEffect()`, especially since it's a setState.

Reply:
Do you mean to replace the `reportingState` by a `useMemo` ?
But there is a `onClick` below : 
```
onClick={() => setReportingTab(ReportingTab.CURRENT_REPORTING)}
```